Mathlib/Analysis/RCLike/Inner.lean

```diff
@@ -0,0 +1,150 @@
+/-
+Copyright (c) 2023 Yaël Dilies. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dilies
+-/
+import Mathlib.Analysis.InnerProductSpace.PiL2
+
+/-!
+# L2 inner product of finite sequences
+
+This file defines the weighted L2 inner product of functions `f g : ι → R` where `ι` is a fintype as
+`∑ i, conj (f i) * g i`. This convention (conjugation on the left) matches the inner product coming
+from `RCLike.innerProductSpace`.
+-/
+
+open Finset Function Real
+open scoped BigOperators ComplexConjugate ENNReal NNReal NNRat
+
+variable {ι κ 𝕜 : Type*} {E : ι → Type*} [Fintype ι]
+
+namespace RCLike
+variable [RCLike 𝕜]
+
+section Pi
+variable [∀ i, SeminormedAddCommGroup (E i)] [∀ i, InnerProductSpace 𝕜 (E i)] {w : ι → ℝ}
+
+/-- Weighted inner product giving rise to the L2 norm. -/
+def wInner (w : ι → ℝ) (f g : ∀ i, E i) : 𝕜 := ∑ i, w i • inner (f i) (g i)
+
+/-- The weight function making `wInner` into the compact inner product. -/
+noncomputable abbrev cWeight : ι → ℝ := Function.const _ (Fintype.card ι)⁻¹
```

Comment:
I'm not sure about the name here. `RCLike.cWeight` doesn't seem very descriptive. And it doesn't seem to be connected meaningfully to the notation below.

Reply:
What would you suggest instead?

There might an option of writing a more complicated elaborator that figures `ι` out from `f`, but I haven't written such a complicated thing yet.

Reply:
I don't know, which is why I didn't suggest anything before. Maybe `cardWeight`? I'm not going to force it though.

Reply:
`c` stands for "compact", not "card"